jcommune-service/src/main/java/org/jtalks/jcommune/service/transactional/TransactionalAuthenticator.java

```diff
@@ -341,7 +341,12 @@ public BindingResult register(RegisterUserDto registerUserDto)
         }
 
         if(result.hasErrors()) {
-            userDto.setPassword(notEncodedPassword);
+            if (wrongPasswordConfirmError(result)) {
+                removePasswords(registerUserDto);
```

Comment:
Don't think that it's correct to mix layers. `TransactionalAuthenticator` wouldn't know about form. Proposed method name just for programmer and can tangle.